### PR TITLE
Convert LSP6 tests to Chai

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

## 🧪 Tests

- [x] Convert all the tests from the LSP6 test suite from Jest > Chai, using the `expect` assertion library from Chai. 

This includes the following changes:
    -  `toEqual` > `to.equal` (for single values) + `to.deep.equal` (for asserting arrays).
    - `toBeRevertedWith(...)` > `to.be.revertedWith(...)` and `to.be.revertedWithCustomError(...).withArgs(...)`
    - `toBeTruthy()` > `to.be.true;`
    - `beforeAll` hooks with `before`
    - `it.each` with `Array.forEach` syntax
    - in tests for permission `TRANSFERLYX`, use Hardhat Chai matcher [`.to.changeEthersBalances([...], [...])`]() to test balance changes between sending UP and recipient 🔧 

- [x] create the helper function `combinePermissions()`, to be used in tests to combine multiple permissions in one `bytes32` hex value.

## 🏗️ Build

- [x] edit the `npm run test:lsp6` script to run tests with Hardhat + Chai ✅ , and not with Jest anymore ❌

### CODE REVIEW

**Review by YamenMerhi — APPROVED** [review]
Cool! 
`CombinedPermissions(,,)` function is very useful, as it's shared across all the tests.

File: tests/LSP6KeyManager/tests/AllowedFunctions.test.ts [hunk 1]
```diff
@@ -104,13 +102,8 @@ export const shouldBehaveLikeAllowedFunctions = (
...
 
           let result = await targetContract.callStatic.getNumber();
-          expect(
-            parseInt(ethers.BigNumber.from(result).toNumber(), 10) !==
-              ethers.BigNumber.from(initialNumber).toNumber()
-          );
-          expect(
-            parseInt(ethers.BigNumber.from(result).toNumber(), 10)
-          ).toEqual(newNumber);
+          expect(result.toNumber()).to.not.equal(initialNumber.toNumber());
```
Comment (b00ste) [inline tests/LSP6KeyManager/tests/AllowedFunctions.test.ts:105]: No need for `.toNumber()`. Test is passing without that conversion

File: tests/LSP6KeyManager/tests/AllowedFunctions.test.ts [hunk 2]
```diff
@@ -104,13 +102,8 @@ export const shouldBehaveLikeAllowedFunctions = (
...
+          expect(result.toNumber()).to.equal(newNumber);
```
Comment (b00ste) [inline tests/LSP6KeyManager/tests/AllowedFunctions.test.ts:106]: No need for `.toNumber()`. Test is passing without that conversion

File: tests/LSP6KeyManager/tests/AllowedFunctions.test.ts [hunk 3]
```diff
@@ -166,20 +154,19 @@ export const shouldBehaveLikeAllowedFunctions = (
...
+            );
 
           let result = await targetContract.callStatic.getNumber();
-          expect(
-            parseInt(ethers.BigNumber.from(result).toNumber(), 10) !== newNumber
-          );
-          expect(
-            parseInt(ethers.BigNumber.from(result).toNumber(), 10)
-          ).toEqual(ethers.BigNumber.from(initialNumber).toNumber());
+          expect(result.toNumber()).to.not.equal(newNumber);
```
Comment (b00ste) [inline tests/LSP6KeyManager/tests/AllowedFunctions.test.ts:168]: Same as above

File: tests/LSP6KeyManager/tests/AllowedFunctions.test.ts [hunk 4]
```diff
@@ -166,20 +154,19 @@ export const shouldBehaveLikeAllowedFunctions = (
...
+          expect(result.toNumber()).to.equal(initialNumber.toNumber());
```
Comment (b00ste) [inline tests/LSP6KeyManager/tests/AllowedFunctions.test.ts:169]: Same as above

**Review by b00ste — COMMENTED** [review]
Everything else looks good

**Pull request merged**